### PR TITLE
Bluetooth: tests: bsim: host/adv/resume2

### PR DESCRIPTION
**Pull request by alwa-nordic**
This test is an attempt at formalizing at least part of the behavior described in commit https://github.com/zephyrproject-rtos/zephyr/commit/372c8f2d92fce3653f5024261e17012197acc330.

### CODE REVIEW

File: tests/bluetooth/common/testlib/include/testlib/conn.h [hunk 1]
```diff
@@ -105,4 +105,18 @@ void bt_testlib_conn_unref(struct bt_conn **connp);
  */
 struct bt_conn *bt_testlib_conn_unindex(enum bt_conn_type conn_type, uint8_t conn_index);
 
+/**
+ * @brief Wait until there is a free connection slot
+ *
+ * Thread-safe.
+ *
+ * Returns when there already is a free conenction slot or a
```
Comment (jori-nordic) [inline tests/bluetooth/common/testlib/include/testlib/conn.h:131]: connection

File: tests/bsim/bluetooth/host/adv/resume2/connecter/main.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#include <errno.h>
+#include <testlib/conn.h>
+#include <testlib/scan.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+LOG_MODULE_REGISTER(connecter, LOG_LEVEL_INF);
+
+int main(void)
```
Comment (jori-nordic) [inline tests/bsim/bluetooth/host/adv/resume2/connecter/main.c:15]: maybe a small description of what is the job of this app (and the connectable) would be good.
I know code is doc and all that, still simpler to just read a comment at the top level

File: tests/bsim/bluetooth/host/adv/resume2/dut/Testing/20240318-1320/Test.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jori-nordic) [inline tests/bsim/bluetooth/host/adv/resume2/dut/Testing/20240318-1320/Test.xml:1]: remove `Testing` from commit?

File: tests/bsim/bluetooth/host/adv/resume2/dut/main.c [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+	LOG_INF("💣 Connect");
+	err = bt_testlib_connect(&connectable_addr, &conn);
+	if (err) {
+		/* If the test fails, it's because the advertiser 'stole' the
+		 * central's connection slot.
+		 */
+		__ASSERT_NO_MSG(err == -ENOMEM);
+
+		LOG_ERR("💥 Advertiser stole the connection slot");
```
Comment (jori-nordic) [inline tests/bsim/bluetooth/host/adv/resume2/dut/main.c:198]: internal: we should do a last test run to see if it still fails with `372c8f2d92` reverted.

**Review by jori-nordic — CHANGES_REQUESTED** [review]

File: tests/bsim/bluetooth/host/adv/resume2/dut/main.c (hunk 4 above)
Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/adv/resume2/dut/main.c:198]: Verified. Fails with `💥 Advertiser stole the connection slot`.

**Review by jori-nordic — DISMISSED** [review]

File: tests/bsim/bluetooth/host/adv/resume2/connecter/main.c (hunk 2 above)
Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/adv/resume2/connecter/main.c:28]: Added

File: tests/bsim/bluetooth/host/adv/resume2/connecter/main.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * This app scans for a device with the name "dut" and connects
+ * to it. It then waits for the connection to be disconnected,
+ * before starting over.
+ *
+ * This app is added to the simulation simply to excerise the
+ * the DUT's connectable advertiser.
+ *
+ * Mulitple instances of this app are added to the simulation,
+ * to excericse BT_MAX_CONN of the DUT.
```
Comment (jori-nordic) [inline tests/bsim/bluetooth/host/adv/resume2/connecter/main.c:25]: you need to exorcise this word (twice lol)

Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/adv/resume2/connecter/main.c]: I have now invested in a spell checker 🧠

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

File: tests/bsim/bluetooth/host/adv/resume2/dut/main.c [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+	bst_result = In_progress;
+
+	err = bt_enable(NULL);
+	__ASSERT_NO_MSG(!err);
+
+	err = bt_set_name("dut");
+	__ASSERT_NO_MSG(!err);
+
+
+	LOG_INF("👉 Preflight test: Advertiser fills connection capacity.");
```
Comment (aescolar) [inline tests/bsim/bluetooth/host/adv/resume2/dut/main.c:86]: Guys this is bonkers... this file is full of emojis... I would have expected we would only accept ascii chars in source.

File: tests/bsim/bluetooth/host/adv/resume2/run.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+import os
```
Comment (aescolar) [inline tests/bsim/bluetooth/host/adv/resume2/run.py:6]: This is not nice :(
Now we have a completely different way of running for a single test. 
This needs to be maintained by other people (usually it is me who needs to do clean ups when something fundamentally different needs to be supported in all tests..)

**Review by aescolar — COMMENTED** [review]
:(

File: tests/bsim/bluetooth/host/adv/resume2/dut/main.c (hunk 6 above)
Comment (fabiobaltieri) [reply tests/bsim/bluetooth/host/adv/resume2/dut/main.c:86]: 😮

Comment (jori-nordic) [reply tests/bsim/bluetooth/host/adv/resume2/dut/main.c:86]: More context after talking with @aescolar : 
- we have blind contributors, their accessibility SW/HW doesn't support 
- some greybeard contributors use editors w/ no emoji support (or partial, like eclipse)

I'm reworking the bsim docs, I will add a point for that in the "Conventions" section.

**Pull request merged**